stackrox/stackrox
# ROX-14398: Remove "Scope Manager" default Role and PermissionSet

## Description

This change is part of a series with the goal of making the configuration of user access control easier, and of limiting the risk of features not working because of missing permissions.

The goal is to bring the `Role` permission under the umbrella of the `Access` one.

This part focuses on the `Scope Manager` default role and permission set, which was flagged as likely to allow more than it should.
The point is to remove the default role and permission set. These items will only be kept in a different form if they are used in the customer environments.

## Checklist
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

CI should be sufficient for the core functionality

Manual testing:

For each of the test cases, spawn a 4.0 instance, prepare the data, upgrade the Central version to the latest image pushed for the current PR, and check the resulting data.

- Only the default roles, permission sets, groups and API tokens. The `Scope Manager` role and permission sets should be removed
- On top of the default roles and permission sets, a new role referencing the `Scope Manager` permission set should be created. In that case, the default `Scope Manager` role should be removed, and the permission set should remain (renamed to `[DEPRECATED] Scope Manager` and be of imperative source).
- On top of the default roles, permission sets and groups, an auth provider configuration should reference the `Scope Manager` role. In that case, the `Scope Manager` role should be renamed, flagged as imperative and the auth provider rule should reference the renamed role.
- On top of the default roles, permission sets and API Tokens, an AIP Token should reference the `Scope Manager` role. In that case, the `Scope Manager` role should be renamed, flagged as imperative and the auth provider rule should reference the renamed role.
